lib/Test/MockFile.pm

```diff
@@ -1227,7 +1254,10 @@ BEGIN {
     };
 
     *CORE::GLOBAL::opendir = sub(*$) {
-        @_ = _upgrade_barewords(@_) if defined $_[0];
+        if ( defined $_[0] ) {
+            @_ = _upgrade_barewords(@_);
+            shift; # Remove bareword indicator
```

Comment:
LGTM, suggestion
```
( undef, @_ ) = upgrade_barewords(@_); # Remove bareword indicator
```

Reply:
Good idea. I'll provide this in a separate PR.